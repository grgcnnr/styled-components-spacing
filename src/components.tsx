import styled from 'styled-components';
import {ThemedValueOrValueMap} from './types';
import {
  ThemedMarginProps,
  ThemedPaddingProps,
  m,
  mx,
  my,
  mt,
  mr,
  mb,
  ml,
  p,
  px,
  py,
  pt,
  pr,
  pb,
  pl,
  marginProps,
  paddingProps,
} from './themed';

interface InlineProps {
  inline?: boolean;
}

const inlineMixin = ({inline}: InlineProps): string | undefined =>
  (inline && 'display: inline-block;') || undefined;

export type DeprecatedSpacingProps = InlineProps & {
  all?: ThemedValueOrValueMap;
  horizontal?: ThemedValueOrValueMap;
  vertical?: ThemedValueOrValueMap;
  top?: ThemedValueOrValueMap;
  right?: ThemedValueOrValueMap;
  bottom?: ThemedValueOrValueMap;
  left?: ThemedValueOrValueMap;
};

/** @deprecated */
export const Margin = styled.div<DeprecatedSpacingProps>`
  ${inlineMixin}
  ${({all}) => all && m(all)}
  ${({horizontal}) => horizontal && mx(horizontal)}
  ${({vertical}) => vertical && my(vertical)}
  ${({top}) => top && mt(top)}
  ${({right}) => right && mr(right)}
  ${({bottom}) => bottom && mb(bottom)}
  ${({left}) => left && ml(left)}
`;

Margin.displayName = 'Margin';

/** @deprecated */
export const Padding = styled.div<DeprecatedSpacingProps>`
  ${inlineMixin}
  ${({all}) => all && p(all)}
  ${({horizontal}) => horizontal && px(horizontal)}
  ${({vertical}) => vertical && py(vertical)}
  ${({top}) => top && pt(top)}
  ${({right}) => right && pr(right)}
  ${({bottom}) => bottom && pb(bottom)}
  ${({left}) => left && pl(left)}
`;

Padding.displayName = 'Padding';

export type SpacingProps = InlineProps & ThemedMarginProps & ThemedPaddingProps;

export const Spacing = styled.div<SpacingProps>`
  ${inlineMixin}
  ${marginProps}
  ${paddingProps}
`;

Spacing.displayName = 'Spacing';
