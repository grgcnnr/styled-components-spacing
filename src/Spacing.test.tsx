import 'jest-styled-components';
import * as React from 'react';
import {render} from '@testing-library/react';
import {Spacing} from './components';

describe('<Spacing/>', () => {
  it('it should have no styles when no props are passed', () => {
    const {container} = render(<Spacing />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('inline', () => {
    it('it should be set when the prop is true', () => {
      const {container} = render(<Spacing inline={true} />);
      expect(container.firstChild).toMatchSnapshot();
      expect(container.firstChild).toHaveStyleRule('display', 'inline-block');
    });

    it('it should not be set when the prop is false', () => {
      const {container} = render(<Spacing inline={false} />);
      expect(container.firstChild).not.toHaveStyleRule(
        'display',
        'inline-block',
      );
    });
  });

  describe('margin', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing margin={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing margin={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing margin={{tablet: 2, desktop: 4}} />);
      expect(container.firstChild).toMatchSnapshot();
      expect(container.firstChild).toHaveStyleRule('margin', '0.5rem', {
        media: 'screen and (min-width:46.0625em)',
      });
      expect(container.firstChild).toHaveStyleRule('margin', '2rem', {
        media: 'screen and (min-width:74.6875em)',
      });
    });
  });

  describe('marginX', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginX={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginX={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginX={{tablet: 2, desktop: 4}} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('marginY', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginY={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginY={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginY={{tablet: 2, desktop: 4}} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('marginTop', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginTop={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginTop={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginTop={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('marginRight', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginRight={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginRight={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginRight={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('marginBottom', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginBottom={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginBottom={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginBottom={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('marginLeft', () => {
    it('it should have a single margin for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing marginLeft={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginLeft={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different margin for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing marginLeft={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('padding', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing padding={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing padding={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing padding={{tablet: 2, desktop: 4}} />);
      expect(container.firstChild).toMatchSnapshot();
      expect(container.firstChild).toHaveStyleRule('padding', '0.5rem', {
        media: 'screen and (min-width:46.0625em)',
      });
      expect(container.firstChild).toHaveStyleRule('padding', '2rem', {
        media: 'screen and (min-width:74.6875em)',
      });
    });
  });

  describe('paddingX', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingX={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingX={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingX={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('paddingY', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingY={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingY={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingY={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('paddingTop', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingTop={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingTop={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingTop={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('paddingRight', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingRight={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingRight={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingRight={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('paddingBottom', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingBottom={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingBottom={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingBottom={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('paddingLeft', () => {
    it('it should have a single padding for all breakpoints when the prop is passed', () => {
      const {container} = render(<Spacing paddingLeft={1} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for all breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingLeft={{mobile: 0, tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('it should have a different padding for some breakpoints when the prop is passed', () => {
      const {container} = render(
        <Spacing paddingLeft={{tablet: 2, desktop: 4}} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
