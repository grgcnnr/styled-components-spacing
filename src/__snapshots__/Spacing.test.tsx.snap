// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spacing/> inline it should be set when the prop is true 1`] = `
.c0 {
  display: inline-block;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> it should have no styles when no props are passed 1`] = `
<div
  class=""
/>
`;

exports[`<Spacing/> margin it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> margin it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> margin it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginBottom it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginBottom it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginBottom it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginLeft it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-left: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginLeft it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginLeft it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-left: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginRight it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginRight it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginRight it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginTop it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-top: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginTop it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginTop it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-top: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginX it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-left: 0;
  margin-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-left: 2rem;
    margin-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginX it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-left: 2rem;
    margin-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginX it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginY it should have a different margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginY it should have a different margin for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> marginY it should have a single margin for all breakpoints when the prop is passed 1`] = `
.c0 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> padding it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> padding it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> padding it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingBottom it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingBottom it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingBottom it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingLeft it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingLeft it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingLeft it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingRight it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingRight it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingRight it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingTop it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingTop it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingTop it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingX it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0;
  padding-right: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingX it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingX it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingY it should have a different padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0;
  padding-bottom: 0;
}

@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingY it should have a different padding for some breakpoints when the prop is passed 1`] = `
@media screen and (min-width:46.0625em) {
  .c0 {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
  }
}

@media screen and (min-width:74.6875em) {
  .c0 {
    padding-top: 2rem;
    padding-bottom: 2rem;
  }
}

<div
  class="c0"
/>
`;

exports[`<Spacing/> paddingY it should have a single padding for all breakpoints when the prop is passed 1`] = `
.c0 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

<div
  class="c0"
/>
`;
